pydantic/_internal/_schema_generation_shared.py

```diff
@@ -124,3 +125,49 @@ def resolve_ref_schema(self, maybe_ref_schema: core_schema.CoreSchema) -> core_s
         elif maybe_ref_schema['type'] == 'definitions':
             return self.resolve_ref_schema(maybe_ref_schema['schema'])
         return maybe_ref_schema
+
+
+def get_schema_or_set_mocks(
```

Comment:
I suppose we could use overloads here to be more specific, but that doesn't feel super valuable for this case? Wdyt @Viicos?